### PR TITLE
WIP - Support new Symfony authenticator system

### PR DESCRIPTION
**Pull request by mcsky**
### Changes

This PR add a new authenticator following[ Symfony > 5.1 guidelines](https://symfony.com/doc/current/security/custom_authenticator.html#security-passports). 
It shouldn't impact the current usage of this library because it requires a specific configuration to use it (see update in readme)

### References
This issue is reported [here](https://github.com/auth0/jwt-auth-bundle/issues/115)

### Testing

Use Symfony >5.1 and follow the `config/security.yaml` example in readme file.

- [x] This change adds test coverage
- [x] This change has been tested on the latest version of Symfony

## This PR is in WIP because some points need to be discussed 

1. The data given to the user provider is **no more** a jwt object, it's the JWT string. It's due to the new Symfony mechanism that [allows only a string](https://github.com/symfony/security-http/blob/5.3/Authenticator/Passport/Badge/UserBadge.php#L46). I'm pretty sure it's not possible to work with something else than a string. I spend 1 hour trying to hack this system ^^ 
But I think it's ok that your library API change (jwt object -> string jwt), because it will require a manual change in the configuration to activate this new authenticator. Being explicit in the upgrade.md should be enough 

2. I broke the old tests, but I can't figure out what's the problem. Maybe you could help me @evansims ^^' ?

### CODE REVIEW

**Comment by mcsky** [comment]
@evansims could you add the hacktoberfest label on this PR please 😂 🙏

**Comment by mcsky** [comment]
hey @evansims ! Could you find some time to give your feedback on this? 🙏

**Comment by Nek-** [comment]
@evansims this PR looks good did you manage to have some minutes to take a look at it?

File: composer.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   "require-dev": {
     "phpunit/phpunit": "^9.3",
     "nyholm/symfony-bundle-test": "^1.0.2",
-    "symfony/security": "^4.4 || ~5.1",
+    "symfony/security-http": "^4.4 || ^5.1",
```
Comment (nanofelis) [inline composer.json:29]:
```suggestion
    "symfony/security-http": "^4.4 || ^5.1 || ^6.0",
```

**Pull request merged**